src/pages/journal/journal.ts

```diff
@@ -171,6 +207,70 @@ export class JournalPage extends BasePageComponent implements OnInit {
     }
   }
 
+  /**selectedDate
+   * If state contains no cached data and search hasn't
+   * previously returned results then get users search results
+   * otherwise get create slot data directly
+   * @param emission
+   */
+  generateSlotAndSearchResults = (emission: SlotItem[]) => {
+    if (!this.isCachedTests && this.searchResultsAppRefs.length === 0) {
+      this.getSearchResults(emission);
+    }
+    this.createSlots(emission, this.searchResultsAppRefs);
+  }
+
+  /**
+   * Call endpoint to obtain users previously completed tests and
+   * pass to createSlots function for evaluation
+   * @param emission
+   */
+  getSearchResults = (emission: SlotItem[]) => {
+    const advancedSearchParams: AdvancedSearchParams = {
+      startDate: moment().subtract(14, 'days').format('YYYY-MM-DD'),
+      endDate: moment().format('YYYY-MM-DD'),
+      staffNumber: removeLeadingZeros(this.staffNumber),
+      costCode: '',
+    };
+    return new Promise((resolve) => {
+      this.searchProvider.advancedSearch(advancedSearchParams)
```

Comment:
This should happen in the Journal Effect. Don't think that making the Journal Page dependent on SearchProvider is a good idea.

Reply:
As a matter of fact it should be in its own effect. Let's keep to the Open-closed principle

Reply:
Functionality has been moved to `journal.effects`